keyboards/crkbd/keymaps/kidbrazil/keymap.c

```diff
@@ -54,19 +56,39 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
     )
 };
 
-//int RGB_current_mode;
+// [Post Init] --------------------------------------------------------------//
+void keyboard_post_init_user(void) {
+    // Set RGB to known state
+    rgb_matrix_enable_noeeprom();
+    rgb_matrix_set_color_all(RGB_GREEN);
+    user_led_enabled = true;
 
+}
 // [Process User Input] ------------------------------------------------------//
 bool process_record_user(uint16_t keycode, keyrecord_t *record) {
-    // Use process_record_keymap to reset timer on keypress
-    if (record->event.pressed) {
-        #ifdef OLED_DRIVER_ENABLE
-            oled_timer = timer_read32();
-        #endif
-        // Restore LEDs if they are enabled in eeprom
-        rgb_matrix_enable_noeeprom();
+    switch (keycode) {
+      case RGB_TOG:
+        if (record->event.pressed) {
+            // Toggle matrix on key press
+            user_led_enabled ? rgb_matrix_disable_noeeprom() : rgb_matrix_enable_noeeprom();
+        } else {
+            // Flip User_led_enabled variable on key release
+            user_led_enabled = !user_led_enabled;
+        }
+        return false; // Skip all further processing of this key
+      default:
+          // Use process_record_keymap to reset timer on all other keypresses
+          if (record->event.pressed) {
+              #ifdef OLED_DRIVER_ENABLE
+                  oled_timer = timer_read32();
+              #endif
```

Comment:
There is a logic bug here. Although the variable is named `oled_timer`, it's used to timeout both the RGB Matrix and the OLEDs. As it's written, disabling the OLEDs will break the RGB Matrix timeout because `oled_timer` will not be set. 

To resolve it, I recommend renaming `oled_timer` to `timeout_timer` and remove the conditional `ifdef` here.